hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -674,6 +685,804 @@ public Response getDeletedDirectorySummary() {
+    long bucketId = bucketInfo.getObjectID();
+    if (names.length == 2) {
+      return constructObjectPathWithPrefix(volumeId, bucketId);
+    }
+
+    // Fetch the immediate parentID which could be a directory or the bucket itself
+    BucketHandler handler =
+        getBucketHandler(reconNamespaceSummaryManager, omMetadataManager,
+            reconSCM, bucketInfo);
+    long dirObjectId = handler.getDirInfo(names).getObjectID();
+    return constructObjectPathWithPrefix(volumeId, bucketId, dirObjectId);
+  }
+
+  /**
+   * Common method to retrieve keys from a table based on a search prefix and a limit.
+   *
+   * @param table     The table to retrieve keys from.
+   * @param paramInfo The stats object holds total count, count and limit.
+   * @return A map of keys and their corresponding OmKeyInfo objects.
+   * @throws IOException If there are problems accessing the table.
+   */
+  private Map<String, OmKeyInfo> retrieveKeysFromTable(
+      Table<String, OmKeyInfo> table, ParamInfo paramInfo)
+      throws IOException {
+    boolean skipPrevKey = false;
+    String seekKey = paramInfo.getPrevKey();
+    Map<String, OmKeyInfo> matchedKeys = new LinkedHashMap<>();
+    try (
+        TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>> keyIter = table.iterator()) {
+
+      if (!paramInfo.isSkipPrevKeyDone() && StringUtils.isNotBlank(seekKey)) {
+        skipPrevKey = true;
+        Table.KeyValue<String, OmKeyInfo> seekKeyValue =
+            keyIter.seek(seekKey);
+
+        // check if RocksDB was able to seek correctly to the given key prefix
+        // if not, then return empty result
+        // In case of an empty prevKeyPrefix, all the keys are returned
+        if (seekKeyValue == null ||
+            (StringUtils.isNotBlank(paramInfo.getPrevKey()) &&
+                !seekKeyValue.getKey().equals(paramInfo.getPrevKey()))) {
+          return matchedKeys;
+        }
+      } else {
+        keyIter.seek(paramInfo.getStartPrefix());
+      }
+
+      while (keyIter.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> entry = keyIter.next();
+        String dbKey = entry.getKey();
+        paramInfo.setLastKey(dbKey);
+        if (!dbKey.startsWith(paramInfo.getStartPrefix())) {
+          break; // Exit the loop if the key no longer matches the prefix
+        }
+        if (skipPrevKey && dbKey.equals(paramInfo.getPrevKey())) {
+          paramInfo.setSkipPrevKeyDone(true);
+          continue;
+        }
+        if (applyFilters(entry, paramInfo) && matchedKeys.size() < paramInfo.getLimit()) {
+          matchedKeys.put(dbKey, entry.getValue());
+        }
+      }
+    } catch (IOException exception) {
+      LOG.error("Error retrieving keys from table for path: {}", paramInfo.getStartPrefix(), exception);
+      throw exception;
+    }
+    return matchedKeys;
+  }
+
+  private boolean applyFilters(Table.KeyValue<String, OmKeyInfo> entry, ParamInfo paramInfo) throws IOException {
```

Comment:
Since we are already propagating the `throws IOException` upwards do we require the try and catch block to handle the IOException?

We could further simplify the code in the method by removing them.

Reply:
Yes, we need for defining the predicate definition, else compiler gives error.